### ISSUE TITLE
Core-nav: Oberkategories not seen as active

### ISSUE THREAD

**Issue by onli** [s9y/Serendipity#730]
Normally, when you select a category as target for a menu option, when you visit that category the menu option gets printed without the `a`, it is then marked as active. Skeleton does that for example, using the same logic as 2k11. However, that detection fails when the menu options links to a parent category (a category that has one or multiple subcategories). No menu options is marked as active in that case.

**Comment by yellowled** [comment]
This is prepared in `config.inc.php`:

```php
$serendipity['smarty']->assign(array('currpage'  => "http://".$_SERVER['HTTP_HOST'].$_SERVER['REQUEST_URI'],
                                     'currpage2' => $_SERVER['REQUEST_URI']));
```
and then usually used in `index.tpl`:

```smarty
{if $currpage==$navlink.href or $currpage2==$navlink.href}…{/if}
```
I don't know why or how these would not match up for parent categories, but I don't see how this would be an issue with CSS or templating. (Therefore, I sadly have no idea how to address this.)

**Comment by onli** [comment]
Yeah, I'm really surprised about that bug, and it might be related to something not directly involved. I had hoped the smart code is your work :) Well, we will figure it out.

**Comment by yellowled** [comment]
I mean, the likely solution is that for the URL or a category page of a parent category, neither `currpage` nor `currpage2` is a "correct" match for some odd reason. I have no idea why it would be, though. (Mainly because I don't use child categories and have no clue what makes up their URLs.)

**Comment by onli** [comment]
That's the thing: If I'm not mistaken it's the exact same url. Just a link to a category. That it's a parent/or child category is just a setting in the database.

But maybe I'm missing something, given the code that should be impossible.
